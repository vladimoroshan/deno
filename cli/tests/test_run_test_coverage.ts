Deno.test("spawn test", async function () {
  const process = Deno.run({
    cmd: [
      Deno.execPath(),
      "test",
      "--allow-all",
      "--unstable",
      "test_coverage.ts",
    ],
  });

  await process.status();
  process.close();
});
