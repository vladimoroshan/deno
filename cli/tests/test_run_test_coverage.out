Check [WILDCARD]/$deno$test.ts
running 1 tests
test spawn test ... Check [WILDCARD]/$deno$test.ts
running 1 tests
test returnsFooSuccess ... ok ([WILDCARD])

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out ([WILDCARD])

ok ([WILDCARD])

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out ([WILDCARD])

cover [WILDCARD]/subdir/mod1.ts ... 35.714% (5/14)
   2 | export function returnsHi() {
   3 |     return "Hi";
   4 | }
-----|-----
   8 | export function printHello3() {
   9 |     printHello2();
  10 | }
  11 | export function throwsError() {
  12 |     throw Error("exception from mod1");
  13 | }
cover [WILDCARD]/subdir/print_hello.ts ... 25.000% (1/4)
   1 | export function printHello() {
   2 |     console.log("Hello");
   3 | }
cover [WILDCARD]/subdir/subdir2/mod2.ts ... 62.500% (5/8)
   5 | export function printHello2() {
   6 |     printHello();
   7 | }
cover [WILDCARD]/test_coverage.ts ... 100.000% (5/5)
